lib/rantly/generator.rb

```diff
@@ -156,16 +156,8 @@ def float(distribution = nil, params = {})
     end
   end
 
-  def range(lo = nil, hi = nil)
-    lo ||= INTEGER_MIN
-    hi ||= INTEGER_MAX
-
-    if lo.is_a?(Integer) && hi.is_a?(Integer)
-      rand(hi + 1 - lo) + lo
-    else
-      # Assume floats
-      rand * (hi - lo) + lo
-    end
+  def range(lo = INTEGER_MIN, hi = INTEGER_MAX)
```

Comment:
We could in the future verify the range is valid before calling rand, and if it is not raise an exception so the user knows right away the code is invalid.

Reply:
well, returning `nil` is already feedback for the user, right? I think it is easier to check if a value is `nil` than catching exceptions... :thinking:

Reply:
I'm thinking if this generator is used and the value is passed to another method. It may not be obvious that the generated value is sometimes `nil`, and user probably wouldn't be expecting `nil` so it might go unnoticed. An exception makes it clear the provided inputs are invalid.

Reply:
yes, but it could be incoherent with the other methods, as any of them raise an Exception. In addition, it could happen that you want to use the return value to know if the given range was invalid.

Anyway, as you already approved it, I'll merge it and release the new version. Feel free to open an issue if you want to discuss this further. 

Thanks for the review! :wink: